src/Errors/EncoderError.js

```diff
@@ -0,0 +1,8 @@
+class EncoderError extends Error {
+    constructor(message) {
+        super(message)
+        this.name = 'ENCODER_ERROR'
```

Comment:
```suggestion
        this.name = 'EncoderError'
```
The CamelCase would be consistent with native JS errors:
```js
> (new TypeError('test')).toString()
'TypeError: test' // not 'TYPE_ERROR: test'
```

Reply:
well .. OK, constants like names look much more "custom" to me